### PR TITLE
fix: support for connecting to realtime channel for federated boards

### PR DESCRIPTION
**Pull request by pcl**
Federated boards are ones that live in a cluster other than the active
user's default cluster. Registration of a websocket for these boards
must happen through a channel-aware registration API, rather than the
deprecated cluster-wide registration API.

This change uses that approach when the board.useFederatedRegistration
configuration is set to 'true', and uses the current approach when
that config is not set. So, clients wishing for the federation-capable
approach must set the config option in the config tree when constructing
their Webex object.

This could be further developed to use a feature toggle for rollout, but
any such improvements should still respect the configuration value as a
default, so that clients other than the web client can enable the
federated behavior.

This addresses #2155.

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-board/src/realtime.js [hunk 1]
```diff
@@ -91,19 +91,27 @@ const RealtimeService = WebexPlugin.extend({
...
 
-        return this.realtimeChannels.get(channel.channelId);
+      registrationPromise = this.webex.internal.board.register({bindings: requestBindings})
+        .then(({webSocketUrl, bindings}) => ({socketUrl: webSocketUrl, bindings: bindings[0]}));
+    }
+
+    return registrationPromise.then((registration) => {
+      this.realtimeChannels.add({
+        ...registration,
+        channelId: channel.channelId
```
Comment (mwegman) [inline packages/node_modules/@webex/internal-plugin-board/src/realtime.js:110]: Do you not want the `channelId` from `registration` to override the `channelId` from `channel`?

Comment (pcl) [reply packages/node_modules/@webex/internal-plugin-board/src/realtime.js:110]: The data structure generated by `registrationPromise` doesn't include a `channelId` field.

File: packages/node_modules/@webex/internal-plugin-board/src/board.js [hunk 2]
```diff
@@ -506,6 +506,22 @@ const Board = WebexPlugin.extend({
...
+  registerToChannel(channel) {
+    return this.webex.request({
+      method: 'POST',
+      api: 'board',
+      uri: `${channel.channelUrl}/register`,
+      body: {action: 'ADD'}
+    })
+      .then((res) => res.body);
+  },
+
```
Comment (toretre) [inline packages/node_modules/@webex/internal-plugin-board/src/board.js:524]: @pcl  I wonder if there should be a corresponding method for removing the binding ? Or could clients use 'unregisterFromSharedMercury' method ?

Comment (pcl) [reply packages/node_modules/@webex/internal-plugin-board/src/board.js:524]: Yes, `unregisterFromSharedMercury()` is sufficient from what we've seen.

**Comment by arun3528** [comment]
Can we add some tests here

**Comment by jonniespratley** [comment]
Can you please squash this commit.

**Review by jonniespratley — APPROVED** [review]
Also can we get a unit-test for these changes, and they will be good to go.

**Pull request closed without merging**